common/scripted_triggers/wc_professions_reagent_triggers.txt

```diff
@@ -0,0 +1,219 @@
+﻿reagent_rarity_is_very_common_trigger = {
+	OR = {
+		$REAGENT$ = flag:silver
+		$REAGENT$ = flag:iron
+		$REAGENT$ = flag:mithril
+		$REAGENT$ = flag:thorium
+		$REAGENT$ = flag:cobalt
+		$REAGENT$ = flag:titanium
+		$REAGENT$ = flag:ore
+	}
+}
+
+reagent_rarity_is_common_trigger = {
+	OR = {
+		$REAGENT$ = flag:ghost_iron
+		$REAGENT$ = flag:pyrite
+		$REAGENT$ = flag:trillium
+		$REAGENT$ = flag:leystone
+		$REAGENT$ = flag:earthen
+		$REAGENT$ = flag:felsteel
+	}
+}
+
+reagent_rarity_is_uncommon_trigger = {
+	OR = {
+		$REAGENT$ = flag:elementium
+		$REAGENT$ = flag:titansteel
+		$REAGENT$ = flag:khorium
+		$REAGENT$ = flag:kyparite
+		$REAGENT$ = flag:emerald
+		$REAGENT$ = flag:draconium
+		$REAGENT$ = flag:ruby
+	}
+}
+
+#all gems are bit rare
+reagent_rarity_is_rare_trigger = {
+	OR = {
+		$REAGENT$ = flag:gold
+		$REAGENT$ = flag:mojo_stone
+		$REAGENT$ = flag:arcane_stone
+		$REAGENT$ = flag:nethercite
+		$REAGENT$ = flag:diamond
+		$REAGENT$ = flag:mana_stone
+	}
+}
+
+reagent_rarity_is_exotic_trigger = {
+	OR = {
+		$REAGENT$ = flag:opal
+		$REAGENT$ = flag:amethyst
+		$REAGENT$ = flag:saronite
+		$REAGENT$ = flag:obsidium
+		$REAGENT$ = flag:night_stone
+		$REAGENT$ = flag:lapis_lazuli
+	}
+}
+
+reagent_is_gem_trigger = {
+	OR = {
+		$REAGENT$ = flag:lapis_lazuli
+		$REAGENT$ = flag:night_stone
+		$REAGENT$ = flag:ruby
+		$REAGENT$ = flag:amethyst
+		$REAGENT$ = flag:emerald
+		$REAGENT$ = flag:opal
+		$REAGENT$ = flag:diamond
+		$REAGENT$ = flag:nethercite
+		$REAGENT$ = flag:kyparite
+		$REAGENT$ = flag:mojo_stone 
+	}
+}
+
+is_mining_reagent = {
+	$REAGENT$ = { is_in_list = mining_reagents }
+}
+
+#PROVINCE SCOPE
+province_has_reagent_trigger = {
```

Comment:
i believe that we will have to separate this out into a separate effect. think of how absolutely enormous this will get once we have cooking and herbalism as well.